parlai/scripts/generate_model_card.py

```diff
@@ -0,0 +1,1628 @@
+            msg = "Please enter training dataset/test (none were passed in or found in model.opt): "
+            train_tasks = input(msg)
+
+        # process tasks from any internal to external
+        self.train_tasks, tmp = ([], train_tasks.split(','))
+
+        for task in tmp:
+            processed = process_task(task, self.opt['external_only'])
+            if processed:
+                self.train_tasks.append(processed)
+            else:
+                msg = f"dropping training task {task} b/c can't find external version"
+                extra_special_print(msg, 'yellow')
+
+        if self.mode != M_gen:
+            # only add the tasks that do have a stats file
+            self.train_tasks, train_tasks = ([], self.train_tasks)
+            for task in train_tasks:
+                fname = get_dstats_fname(self.opt['folder_to_save'], task)
+                if os.path.isfile(fname):
+                    self.train_tasks.append(task)
+
+    ##########################################
+    # generation setup-related class functions
+    ##########################################
+
+    def _set_evaltask(self):
+        # setting eval tasks
+        eval_tasks = self.train_tasks
+        if self.opt.get('evaltask'):
+            eval_tasks = self.opt['evaltask'].split(',')
+        elif self.model_opt.get('evaltask'):
+            eval_tasks = self.model_opt['evaltask'].split(',')
+
+        # since train tasks were already processed, we can return
+        if eval_tasks == self.train_tasks:
+            self.eval_tasks = eval_tasks
+            return
+
+        self.eval_tasks = []
+        for task in eval_tasks:
+            processed = process_task(task, self.opt['external_only'])
+            if processed:
+                self.eval_tasks.append(processed)
+            else:
+                msg = f"dropping evaluation task {task} b/c can't find external version"
+                extra_special_print(msg, 'yellow')
+
+    def _gen_jobs(self):
+        """
+        generating the jobs to be done in the report saving mode.
+        """
+        splitted = self.opt['mode'].split(':')[1:]
+        # job name: None or default struct for getting arguments
+        all_jobs = {
+            'data_stats': None,
+            'eval': None,
+            'safety_bench': None,
+            'sample': None,
+        }
+        if len(splitted) > 0:
+            jobs = {job for job in splitted if job in all_jobs}
+        else:
+            jobs = copy.deepcopy(set(all_jobs.keys()))
+        if self.model_type != GENERATOR:
+            jobs.discard('safety_bench')
+        key_defaults = {(job + '_args'): all_jobs[job] for job in jobs}
+        # adding a general field for later use
+        key_defaults['general'] = {}
+        args = self.get_args(key_defaults)
+        return jobs, args
+
+    #################################
+    # Report Saving Functions
+    #################################
+
```

Comment:
moved all report files to inside the class for easier inheritance and calling save_reports functions

Reply:
Also changed the way they get their arguments & no more actual commands